pglookout/cluster_monitor.py

```diff
@@ -295,6 +301,53 @@ def update_cluster_member_state(self, instance, db_conn):
         else:
             self.cluster_state[instance] = result
 
+        repl_slots_info = self.cluster_state[instance].get("replication_slots")
+        if repl_slots_info:
+            # This is the master, add new state to the cache
+            for repl_slot in repl_slots_info:
+                self.replication_slots_cache.setdefault(repl_slot["slot_name"], [])
+                slot_states = self.replication_slots_cache[repl_slot["slot_name"]]
+                # If cache is full or client did not flush yet, or we have this LSN already - skip it
+                if len(slot_states) >= self.replication_slots_cache_size or repl_slot["confirmed_flush_lsn"] is None or \
+                    next(
+                        (slot_state for slot_state in slot_states
+                            if slot_state["confirmed_flush_lsn"] == repl_slot["confirmed_flush_lsn"]), None
+                ):
```

Comment:
I think the different cases handled by that part are not all tested, like the `confirmed_flush_lsn` being None or not.